### PR TITLE
Add option to select other layer directories in create-layer

### PR DESCRIPTION
**Pull request by justbur**
I was also interested in @syl20bnr's suggestion in #1620. 

Sorry for messing up the previous pull request.

### CODE REVIEW

**Comment by syl20bnr** [comment]
Thank you ! Se can do even better by proposing with an helm front-end a list of all the directories contained in `dotspacemacs-configuration-layer-path` and `private` directory. The user can select one of the directory of type a new one.
You can find some example to use the helm front-end here: https://github.com/syl20bnr/spacemacs/blob/develop/contrib/lang/sql/packages.el#L31-L46

**Comment by justbur** [comment]
Wouldn't just using `completing-read` as it is now allow for that and be flexible if someone were using `ido-ubiquitous`? That was the original intent. I'm happy to switch to relying on helm alone if you prefer that. 

By the way, 6690b40 allows for the creation of parent directories (so you can start a new "base" layer directory in this command if you wish).

**Comment by syl20bnr** [comment]
I don't use ido-ubiquitous so I had to type my whole path which is not convenient. That info is available in the `dotspacemacs-configuration-layer-path` so we should make good use of it.
Since Spacemacs tries to unify everything under helm we should use this front-end.

**Comment by justbur** [comment]
Is this what you were thinking of?

File: core/core-configuration-layer.el [hunk 1]
```diff
@@ -119,11 +119,28 @@ sub-directory of the contribution directory.")
...
+  "Ask the user for a configuration layer name and the layer
+directory to use. Create a layer with this name in the selected
+layer directory."
+  (interactive)
+  (let* ((layer-path-sel (helm
+                          :sources (list (configuration-layer//create-layer-source))
+                          :prompt "Configuration Layer Path: "))
+         (layer-path (if (not (file-exists-p layer-path-sel))
+                         (read-file-name "New layer path (new or existing directory): " "~/" )
+                       layer-path-sel))
```
Comment (syl20bnr) [inline core/core-configuration-layer.el]: To be more explicit you can add an entry to the sources (line 137) called `Another directory...` and test `layer-path-sel` against this value.
If `layer-path-sel` is neither of the provided candidates then do nothing and warn about it.

**Comment by justbur** [comment]
@syl20bnr Let me know if you'd like other adjustments.

**Comment by justbur** [comment]
@syl20bnr Can I delete this branch? Looks like you merged it already

**Comment by syl20bnr** [comment]
Oops sorry I forgot to tell you that it is in the box :-)
Thank you for this great PR ! :heart:  It is available in 0.102.0.
You can safely delete your branch.

**Pull request closed without merging**